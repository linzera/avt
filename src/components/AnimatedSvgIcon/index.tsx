import React from 'react';
import {Svg, Path, PathProps} from 'react-native-svg';
import Animated from 'react-native-reanimated';
import colors from '~/theme/colors';

export const chevronD =
  'M16.7777 3.22221C16.3481 2.79263 15.6517 2.79263 15.2221 3.22221L7.22208 11.2222C6.79251 11.6518 6.79251 12.3483 7.22208 12.7778L15.2221 20.7778C15.6517 21.2074 16.3481 21.2074 16.7777 20.7778C17.2073 20.3483 17.2073 19.6518 16.7777 19.2222L9.55554 12L16.7777 4.77784C17.2073 4.34827 17.2073 3.65178 16.7777 3.22221Z';

export const shareD =
  'M12.7808 1.29591C12.5887 1.08302 12.3105 0.949219 12.0012 0.949219C12.0008 0.949219 12.0005 0.949219 12.0002 0.949219C11.7311 0.948974 11.462 1.05149 11.2568 1.25676L7.25676 5.25676C6.84671 5.66681 6.84671 6.33163 7.25676 6.74168C7.66681 7.15173 8.33163 7.15173 8.74168 6.74168L10.9512 4.53219V14.9992C10.9512 15.5791 11.4213 16.0492 12.0012 16.0492C12.5811 16.0492 13.0512 15.5791 13.0512 14.9992V4.53609L15.2568 6.74168C15.6668 7.15173 16.3316 7.15173 16.7417 6.74168C17.1517 6.33163 17.1517 5.66681 16.7417 5.25676L12.7808 1.29591ZM3.99727 10.9512C4.57716 10.9512 5.04727 11.4213 5.04727 12.0012V20.0012C5.04727 20.2531 5.14735 20.4948 5.32551 20.6729C5.50367 20.8511 5.74531 20.9512 5.99727 20.9512H17.9973C18.2492 20.9512 18.4909 20.8511 18.669 20.6729C18.8472 20.4948 18.9473 20.2531 18.9473 20.0012V12.0012C18.9473 11.4213 19.4174 10.9512 19.9973 10.9512C20.5772 10.9512 21.0473 11.4213 21.0473 12.0012V20.0012C21.0473 20.8101 20.7259 21.5859 20.1539 22.1578C19.582 22.7298 18.8062 23.0512 17.9973 23.0512H5.99727C5.18835 23.0512 4.41258 22.7298 3.84059 22.1578C3.2686 21.5859 2.94727 20.8101 2.94727 20.0012V12.0012C2.94727 11.4213 3.41737 10.9512 3.99727 10.9512Z';

const AnimatedPath = Animated.createAnimatedComponent(Path);

interface Props {
  d: string;
  animatedProps?: PathProps;
  fill?: keyof typeof colors;
}

export default function AnimatedSvgIcon(props: Props) {
  return (
    <Svg width={24} height={24} viewBox="0 0 24 24" fill="none">
      <AnimatedPath
        d={props.d}
        animatedProps={{initial: [], ...props.animatedProps}}
        fillRule="evenodd"
        clipRule="evenodd"
        fill={props.fill ? colors[props.fill] : colors.primary}
      />
    </Svg>
  );
}
