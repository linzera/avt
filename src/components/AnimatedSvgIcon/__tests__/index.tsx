import React from 'react';
import {render} from '@testing-library/react-native';
import AnimatedSvgIcon, {chevronD, shareD} from '..';

describe('<AnimatedSvgIcon />', () => {
  it('should render chevron as expected', () => {
    const {toJSON} = render(<AnimatedSvgIcon d={chevronD} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('should render share as expected', () => {
    const {toJSON} = render(<AnimatedSvgIcon d={shareD} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
