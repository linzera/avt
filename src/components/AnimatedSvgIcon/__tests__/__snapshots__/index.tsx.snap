// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimatedSvgIcon /> should render chevron as expected 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={24}
  bbWidth={24}
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      Array [
        "fill",
      ]
    }
  >
    <RNSVGPath
      clipRule={0}
      d="M16.7777 3.22221C16.3481 2.79263 15.6517 2.79263 15.2221 3.22221L7.22208 11.2222C6.79251 11.6518 6.79251 12.3483 7.22208 12.7778L15.2221 20.7778C15.6517 21.2074 16.3481 21.2074 16.7777 20.7778C17.2073 20.3483 17.2073 19.6518 16.7777 19.2222L9.55554 12L16.7777 4.77784C17.2073 4.34827 17.2073 3.65178 16.7777 3.22221Z"
      fill={4278332244}
      fillRule={0}
      propList={
        Array [
          "fill",
          "fillRule",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`<AnimatedSvgIcon /> should render share as expected 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={24}
  bbWidth={24}
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      Array [
        "fill",
      ]
    }
  >
    <RNSVGPath
      clipRule={0}
      d="M12.7808 1.29591C12.5887 1.08302 12.3105 0.949219 12.0012 0.949219C12.0008 0.949219 12.0005 0.949219 12.0002 0.949219C11.7311 0.948974 11.462 1.05149 11.2568 1.25676L7.25676 5.25676C6.84671 5.66681 6.84671 6.33163 7.25676 6.74168C7.66681 7.15173 8.33163 7.15173 8.74168 6.74168L10.9512 4.53219V14.9992C10.9512 15.5791 11.4213 16.0492 12.0012 16.0492C12.5811 16.0492 13.0512 15.5791 13.0512 14.9992V4.53609L15.2568 6.74168C15.6668 7.15173 16.3316 7.15173 16.7417 6.74168C17.1517 6.33163 17.1517 5.66681 16.7417 5.25676L12.7808 1.29591ZM3.99727 10.9512C4.57716 10.9512 5.04727 11.4213 5.04727 12.0012V20.0012C5.04727 20.2531 5.14735 20.4948 5.32551 20.6729C5.50367 20.8511 5.74531 20.9512 5.99727 20.9512H17.9973C18.2492 20.9512 18.4909 20.8511 18.669 20.6729C18.8472 20.4948 18.9473 20.2531 18.9473 20.0012V12.0012C18.9473 11.4213 19.4174 10.9512 19.9973 10.9512C20.5772 10.9512 21.0473 11.4213 21.0473 12.0012V20.0012C21.0473 20.8101 20.7259 21.5859 20.1539 22.1578C19.582 22.7298 18.8062 23.0512 17.9973 23.0512H5.99727C5.18835 23.0512 4.41258 22.7298 3.84059 22.1578C3.2686 21.5859 2.94727 20.8101 2.94727 20.0012V12.0012C2.94727 11.4213 3.41737 10.9512 3.99727 10.9512Z"
      fill={4278332244}
      fillRule={0}
      propList={
        Array [
          "fill",
          "fillRule",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
